modules/frontend/hedged_requests.go

```diff
@@ -0,0 +1,53 @@
+package frontend
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/cristalhq/hedgedhttp"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+const (
+	hedgedMetricsPublishDuration = 10 * time.Second
+)
+
+var (
+	hedgedRequestsMetrics = promauto.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "tempo",
+			Name:      "query_frontend_hedged_roundtrips_total",
+			Help:      "Total number of hedged search requests. Registered as a gauge for code sanity. This is a counter.",
+		},
+	)
+)
+
+func newHedgedRequestWare(cfg HedgingConfig) Middleware {
+	return MiddlewareFunc(func(next http.RoundTripper) http.RoundTripper {
+		if cfg.HedgeRequestsAt == 0 {
+			return next
+		}
+		ret, stats, err := hedgedhttp.NewRoundTripperAndStats(cfg.HedgeRequestsAt, cfg.HedgeRequestsUpTo, next)
+		if err != nil {
+			panic(err)
```

Comment:
should we just 500 the call?

Reply:
Hmm, this is a invalid config, I'd rather that Tempo doesn't start.